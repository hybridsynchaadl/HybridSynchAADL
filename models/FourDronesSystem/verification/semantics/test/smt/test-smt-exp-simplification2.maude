set show timing off .

load ../../interpreter-symbolic2.maude

select SMT-CHECKSAT .

red simplify(b(0) and ([false] or b(0))) .
red simplify([true] and b(2) or (b(4) === [false])) .
red simplify(b(0) === b(0) ? [true] and b(2) : b(3)) .
red simplify([true]) .
red simplify(not [true]) .
red simplify([true] and [false]) .
red simplify([true] xor [false]) .
red simplify([true] or [false]) .
red simplify([true] implies [false]) .
red simplify([true] === [false]) .
red simplify([true] =/== [false]) .

red simplify(([3/1] * [3/4]) < r(0)) .
red simplify(r(1) > r(1) ? [2/1] / [5/1] : [1/1] * [3/1] + [6/1]) .
red simplify([1/1] + [2/1] + [3/1] + (- [2/1])) .
red simplify([1/1] + [1/1]) .
red simplify([1/1] * [1/1]) .
red simplify([1/1] - [1/1]) .
red simplify([1/1] / [1/1]) .

red simplify([1/1] < [1/1]) .
red simplify([1/1] <= [1/1]) .
red simplify([1/1] > [1/1]) .
red simplify([1/1] >= [1/1]) .
