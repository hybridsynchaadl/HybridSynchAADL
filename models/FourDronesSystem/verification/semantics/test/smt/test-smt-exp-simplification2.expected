==========================================
reduce in SMT-CHECKSAT : simplify(b(0) and (b(0) or [(false).Boolean])) .
rewrites: 10
result SMTBoolVar: [b("#$#0")]
==========================================
reduce in SMT-CHECKSAT : simplify(b(2) and [(true).Boolean] or b(4) === [(
    false).Boolean]) .
rewrites: 17
result SMTBoolExp: [b("#$#2") or not b("#$#4")]
==========================================
reduce in SMT-CHECKSAT : simplify(b(0) === b(0) ? b(2) and [(true).Boolean] :
    b(3)) .
rewrites: 23
result SMTBoolVar: [b("#$#2")]
==========================================
reduce in SMT-CHECKSAT : simplify([(true).Boolean]) .
rewrites: 2
result SMTBoolValue: [(true).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify(not [(true).Boolean]) .
rewrites: 3
result SMTBoolValue: [(false).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(true).Boolean] and [(false).Boolean]) .
rewrites: 3
result SMTBoolValue: [(false).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(true).Boolean] xor [(false).Boolean]) .
rewrites: 3
result SMTBoolValue: [(true).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(true).Boolean] or [(false).Boolean]) .
rewrites: 3
result SMTBoolValue: [(true).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(true).Boolean] implies [(false).Boolean]) .
rewrites: 3
result SMTBoolValue: [(false).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(true).Boolean] === [(false).Boolean]) .
rewrites: 3
result SMTBoolValue: [(false).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(true).Boolean] =/== [(false).Boolean]) .
rewrites: 3
result SMTBoolValue: [(true).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(3/4).Real] * [(3/1).Real] < r(0)) .
rewrites: 10
result SMTBoolExp: [not r("#$#0") <= (9/4).Real]
==========================================
reduce in SMT-CHECKSAT : simplify(r(1) > r(1) ? [(2/1).Real] / [(5/1).Real] : [
    (6/1).Real] + [(1/1).Real] * [(3/1).Real]) .
rewrites: 13
result SMTRealValue: [(9/1).Real]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] + [(2/1).Real] + - [(2/1).Real]
    + [(3/1).Real]) .
rewrites: 6
result SMTRealValue: [(4/1).Real]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] + [(1/1).Real]) .
rewrites: 3
result SMTRealValue: [(2/1).Real]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] * [(1/1).Real]) .
rewrites: 3
result SMTRealValue: [(1/1).Real]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] - [(1/1).Real]) .
rewrites: 3
result SMTRealValue: [(0/1).Real]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] / [(1/1).Real]) .
rewrites: 3
result SMTRealValue: [(1/1).Real]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] < [(1/1).Real]) .
rewrites: 3
result SMTBoolValue: [(false).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] <= [(1/1).Real]) .
rewrites: 3
result SMTBoolValue: [(true).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] > [(1/1).Real]) .
rewrites: 3
result SMTBoolValue: [(false).Boolean]
==========================================
reduce in SMT-CHECKSAT : simplify([(1/1).Real] >= [(1/1).Real]) .
rewrites: 3
result SMTBoolValue: [(true).Boolean]
Bye.
